dozer/cogs/roles.py

```diff
@@ -221,7 +221,16 @@ async def giveme(self, ctx, *, roles):
             e.add_field(name='{} role(s) could not be found!'.format(extra),
                         value='Use `{0.prefix}{0.invoked_with} list` to find valid giveable roles!'.format(ctx),
                         inline=False)
-        await ctx.send(embed=e)
+        msg = await ctx.send(embed=e)
+        await msg.add_reaction("❌")
+        try:
+            await self.bot.wait_for('reaction_add', timeout=30, check=lambda reaction, reactor:
+                                    reaction.emoji == "❌" and reactor == ctx.author and reaction.message == msg)
+            await msg.delete()
+            await ctx.message.delete()
+        except asyncio.TimeoutError:
+            await msg.clear_reactions()
```

Comment:
These two need to handle the case where the bot doesn't have Manage Message permissions as well.